magma/backend/coreir/coreir_compiler.py

```diff
@@ -64,12 +65,16 @@ def __init__(self, main, basename, opts):
 
     def compile(self):
         result = {}
+        result["symbol_table"] = symbol_table = SymbolTable()
         insert_coreir_wires(self.main)
         insert_wrap_casts(self.main)
         backend = self.backend
         opts = _make_opts(backend, self.opts)
+        opts["symbol_table"] = symbol_table
```

Comment:
Note: there might be a more elegant/proper way to pass this around than in `opts`. (the general mechanism of just attaching it somewhere is fine, but `opts` might be a misleading place to put it, although it is very convenient. not sure on this one.)